File: elpizo/models/items/registry.py
from elpizo.models import items
from elpizo.models.items import restorative
from elpizo.models.items.equipment import feet_items
from elpizo.models.items.equipment import legs_items
from elpizo.models.items.equipment import torso_items
from elpizo.models.items.equipment import weapons


def initialize():
  items.Item.register(torso_items.WhiteLongsleeveShirt)
  items.Item.register(legs_items.TealPants)
  items.Item.register(feet_items.BrownShoes)
  items.Item.register(restorative.Carrot)
  items.Item.register(weapons.Dagger)

File: client/models/items/registry.js
module items from "client/models/items";
module feetItems from "client/models/items/equipment/feetItems";
module legsItems from "client/models/items/equipment/legsItems";
module torsoItems from "client/models/items/equipment/torsoItems";
module weapons from "client/models/items/equipment/weapons";
module restorative from "client/models/items/restorative";
module objects from "client/util/objects";

items.Item.register(torsoItems.WhiteLongsleeveShirt);
items.Item.register(legsItems.TealPants);
items.Item.register(feetItems.BrownShoes);
items.Item.register(weapons.Dagger);
items.Item.register(restorative.Carrot);

export function makeItem(message) {
  if (!objects.hasOwnProp.call(items.Item.REGISTRY, message.type)) {
    return new items.Item(message);
  }

  return new items.Item.REGISTRY[message.type](message);
}
